pkg/resource/replication_group/custom_update_api.go

```diff
@@ -127,14 +127,14 @@ func (rm *resourceManager) modifyReplicationGroup(
 	}
 
 	// Get details using describe cache cluster to compute diff
-	latestCacheCluster, err := rm.describeCacheCluster(ctx, latest)
+	latestCacheCluster, err := rm.describeCacheCluster(ctx, latest.ko)
```

Comment:
This change seems unnecessary... why not just keep the describeCacheCluster function signature the same and "unpack" the `ko` field from within the function body?